DragaliaAPI/Controllers/Dragalia/CharaController.cs

```diff
@@ -660,7 +683,7 @@ CharaUpgradeMaterialTypes isUseSpecialMaterial
         SortedSet<int> nodes = playerCharData.ManaCirclePieceIdList;
         nodes.AddRange(manaNodes);
 
-        if (nodes.Count == 50)
+        if (nodes.Count >= 50 && manaNodes.Select(x => x <= 50).Any())
```

Comment:
```suggestion
        if (nodes.Count >= 50 && manaNodes.Any(x => x <= 50))
```

Reply:
Could skip iterating through manaNodes again by setting a bool at the end of each iteration in the previous loop over the collection like: is50mcUnlockRelevant = is50mcUnlockRelevant && nodeNr <= 50

Reply:
oops, meant: ...|| nodeNr ...